csharp/src/Ice/Communicator-Properties.cs

```diff
@@ -158,8 +158,12 @@ internal PropertyValue(string v, bool u)
                 {
                     pv.Used = true;
 
-                    // Match an integer followed by ms, s, m, or h
-                    Match match = Regex.Match(pv.Val, @"^(\d+)(ms|s|m|h)$");
+                    if (pv.Val == "infinite") {
+                        return TimeSpan.FromMilliseconds(-1);
```

Comment:
Maybe we can use a static readonly like in https://docs.microsoft.com/en-us/dotnet/api/system.threading.timeout.infinitetimespan?view=netcore-3.1

And return always the same object

Reply:
We can just use `Timeout.InfiniteTimeSpan` I guess.

Reply:
Fine with me

Reply:
Should we also add "-1ms" as a value for infinite?
```
if (pv.Val == "infinite" || pv.Val == "-1ms") {
...
```

Reply:
I think just "infinite" is fine